components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/UserAccountRecoveryManager.java

```diff
@@ -656,12 +655,12 @@ private void addRecoveryDataObject(String username, String tenantDomain, String
      *
      * @param username     Username of the user
      * @param tenantDomain Tenant domain of the user
-     * @return User
+     * @return User object.
      */
     private User buildUser(String username, String tenantDomain) {
 
         User user = new User();
-        user.setUserName(UserCoreUtil.removeDomainFromName(username));
+        user.setUserName(username);
```

Comment:
I think it is better to have UserCoreUtil.removeDomainFromName()

Reply:
Fixed: 42b3c10c7f845b81216cd0aaa51d493b4272a74c

Reply:
Fixed: 42b3c10c7f845b81216cd0aaa51d493b4272a74c